src/client/datascience/jupyter/kernels/kernelSelector.ts

```diff
@@ -504,13 +544,18 @@ export class KernelSelector implements IKernelSelectionUsage {
     }
     /**
      * Get our kernelspec and interpreter for a local raw connection
+     *
+     * @param {boolean} [ignoreTrackingKernelInformation]
+     * As a side effect these method tracks the kernel information for telemetry, we should ensure that tracking is disabled for Native Notebooks. Native Notebooks knows exactly what the current kernel information is, webviews/interactive is not the same.
+     * I.e. whenever these methods are called by webviews/interactive assume the return value is the active kernel.
```

Comment:
This seems like the wrong place to track this then? There's an implicit assumption about another piece of code understanding that this is where kernels are tracked whereas the other piece of code is deciding to select them. Does this assumption make this code more fragile?

Reply:
Not sure we can do anything about it, unless we refactor a lot of the existing code. this is how WebViews & Interactive window works today.
When we try to run a cell/code, we do'nt know what the actual kernel is, not upfront (unlike native notebooks). 
In inteacive & webviews, we need to get into the lower layers & that where we try to detect the kernel information. This is something i tried to adddress a few weeks ago, and only got it fixed for Native Notebook.

I'm not too keen on fixing that in this PR, 
its just how our code works today.

Reply:
But agree, its ugly, hence my comments (so we know why its ugly, & don't remove it).

Reply:
Maybe add an issue to fix it?